File: May-LeetCoding-Challenge/08-Check-If-It-Is-A-Straight-Line/Check-If-It-Is-A-Straight-Line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        if(coordinates[1][0]^coordinates[0][0])==0:
            return len(set([x[0] for x in coordinates]))^1==0
        else:
            k = (coordinates[1][1]-coordinates[0][1])/(coordinates[1][0] - coordinates[0][0])
            for i in range(1,len(coordinates)-1):
                if (coordinates[i+1][0] - coordinates[i][0])^0==0:
                    return False
                k1 = (coordinates[i+1][1]-coordinates[i][1])/(coordinates[i+1][0] - coordinates[i][0])
                #print(coordinates[i],k,k1)
                if(k!=k1):
                    return False
            return True

File: May-LeetCoding-Challenge/08-Check-If-It-Is-A-Straight-Line/Check-If-It-Is-A-Straight-Line.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        if(coordinates.length <= 2) return true;
        
        float slope = getSlope(coordinates[0], coordinates[1]);
        float newSlope = 0;
        
        for (int i = 2; i < coordinates.length; i++) {
            newSlope = getSlope(coordinates[0], coordinates[i]);
            
            if(newSlope != slope) return false;
        }
        
        return true;
    }
    
    private float getSlope(int[] p1, int[] p2) {
        if(p2[0] - p1[0] == 0) return 0;
        return (float)(p2[1] - p1[1]) / (p2[0] - p1[0]);
    }
}